From c2607302b5a1dfa1d973f2ea53c1ebaba43638f2 Mon Sep 17 00:00:00 2001
From: Kentaro Hayashi <kenhys@gmail.com>
Date: Tue, 17 Jun 2025 17:30:17 +0900
Subject: [PATCH 12/21] v4l2-gst: Load libv4l2.so.0 instead of libv4l2.so

SO version should be added to keep binary compatibility, and to work
without development package.

In addition, don't use absolute path so that it can be searched from
standard library paths.

Upstream-Status: Pending
Ported from https://github.com/clear-code/meta-browser-hwdecode/blob/try-rzg2l-v4.0-support/recipes-browser/chromium/chromium_109.0.5414/1003-Load-libv4l2.so.0-instead-of-libv4l2.so.patch

Signed-off-by: Kentaro Hayashi <hayashi@clear-code.com>
Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 media/gpu/v4l2/v4l2_device.cc | 7 +------
 media/gpu/v4l2/v4l2_utils.cc  | 7 +------
 2 files changed, 2 insertions(+), 12 deletions(-)

diff --git a/media/gpu/v4l2/v4l2_device.cc b/media/gpu/v4l2/v4l2_device.cc
index 75b820fea0..219a661e61 100644
--- a/media/gpu/v4l2/v4l2_device.cc
+++ b/media/gpu/v4l2/v4l2_device.cc
@@ -48,12 +48,7 @@ using media_gpu_v4l2::InitializeStubs;
 using media_gpu_v4l2::kModuleV4l2;
 using media_gpu_v4l2::StubPathMap;
 
-inline static constexpr char kLibV4l2Path[] =
-#if defined(__aarch64__)
-      "/usr/lib64/libv4l2.so";
-#else
-      "/usr/lib/libv4l2.so";
-#endif
+inline static constexpr char kLibV4l2Path[] = "libv4l2.so.0";
 
 namespace media {
 
diff --git a/media/gpu/v4l2/v4l2_utils.cc b/media/gpu/v4l2/v4l2_utils.cc
index c87f4e0cda..89d534bc7d 100644
--- a/media/gpu/v4l2/v4l2_utils.cc
+++ b/media/gpu/v4l2/v4l2_utils.cc
@@ -75,12 +75,7 @@ using media_gpu_v4l2::InitializeStubs;
 using media_gpu_v4l2::kModuleV4l2;
 using media_gpu_v4l2::StubPathMap;
 
-inline static constexpr char kLibV4l2Path[] =
-#if defined(__aarch64__)
-      "/usr/lib64/libv4l2.so";
-#else
-      "/usr/lib/libv4l2.so";
-#endif
+inline static constexpr char kLibV4l2Path[] = "libv4l2.so.0";
 
 static bool use_libv4l2_ = false;
 
-- 
2.43.0

